Find a functionally equivalent code:android.text.Layout.drawText:COMMENT
Method Modifier: public      hidden      
Comment:/**
 * @hide
 */

@@ -41,55 +41,61 @@
             // Draw all leading margin spans.  Adjust left or right according
             // to the paragraph direction of the line.
             final int length = spans.length;
+            boolean useFirstLineMargin = isFirstParaLine;
             for (int n = 0; n < length; n++) {
+                if (spans[n] instanceof LeadingMarginSpan2) {
+                    int count = ((LeadingMarginSpan2) spans[n]).getLeadingMarginLineCount();
+                    int startLine = getLineForOffset(sp.getSpanStart(spans[n]));
+                    // the count that is greatest
+                    if (i < startLine + count) {
+                        useFirstLineMargin = true;
+                        break;
+                    }
+                }
+            }
+            for (int n = 0; n < length; n++) {
                 if (spans[n] instanceof LeadingMarginSpan) {
                     LeadingMarginSpan margin = (LeadingMarginSpan) spans[n];
-                    boolean useFirstLineMargin = isFirstParaLine;
-                    if (margin instanceof LeadingMarginSpan2) {
-                        int count = ((LeadingMarginSpan2) margin).getLeadingMarginLineCount();
-                        int startLine = getLineForOffset(sp.getSpanStart(margin));
-                        useFirstLineMargin = i < startLine + count;
-                    }
                     if (dir == DIR_RIGHT_TO_LEFT) {
                         margin.drawLeadingMargin(canvas, paint, right, dir, ltop, lbaseline, lbottom, buf, start, end, isFirstParaLine, this);
                         right -= margin.getLeadingMargin(useFirstLineMargin);

